### DISCUSSION TITLE
price_to_sqrtp rounding error

### DISCUSSION THREAD

**Discussion by Mattiatore** [Jeiwan/uniswapv3-book#50] [Milestone 1. First Swap]
In the function for price_to_sqrtp I believe it suffers from the math.sqrt function precision and gives an incorrect result, using the library mpmath for python I get a result that agrees with WolframAlpha:
<img width="460" alt="Screenshot 2023-01-01 at 16 23 45" src="https://user-images.githubusercontent.com/47110871/210175977-2a280b2d-99e3-4b17-b019-7bc904ec15ad.png">
instead of the obtained sqrt(P_c) = 5602277097478614198912276234240

**Comment by Jeiwan** [comment]
I use Python not to get precise results but to explain the math without implementing it in Solidity. So, the precision of the calculations doesn't really matter since, in later milestones, all the calculations are eventually implemented in Solidity.
